src/adapter/src/catalog/builtin_table_updates/notice.rs

```diff
@@ -133,14 +156,23 @@ impl CatalogState {
                 Action::SqlStatements(text) => Datum::String(text),
             });
             // push `message_redacted` column
-            packer.push(Datum::String(&notice.message_redacted));
+            packer.push(match notice.message_redacted.as_deref() {
+                Some(message_redacted) => Datum::String(message_redacted),
+                None => Datum::Null,
+            });
             // push `hint_redacted` column
-            packer.push(Datum::String(&notice.hint_redacted));
+            packer.push(match notice.hint_redacted.as_deref() {
+                Some(hint_redacted) => Datum::String(hint_redacted),
```

Comment:
Uncovered

Reply:
That's weird. This _should be_ covered by the tests in `test/sqllogictest/transform/notice/index_key_empty.slt` and `test/sqllogictest/transform/notice/index_too_wide_for_literal_constraints.slt` which select `n.redacted_hint` from `mz_notices n`.